src/client/NuGet.CommandLine.StagingExtensions/StageManagementResource.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Client.Staging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.v3;
+
+namespace NuGet.CommandLine.StagingExtensions
+{
+    public class StageManagementResource : Protocol.Core.Types.INuGetResource
+    {
+        private const string ApiKeyHeader = "X-NuGet-ApiKey";
+        private const string StagePath = "api/stage";
+        private const string MediaType = "application/json";
+
+        private HttpSource _httpSource;
+        private string _stageServiceUri;
+
+        public StageManagementResource(string stageServiceUri, HttpSource httpSource)
+        {
+            if (string.IsNullOrEmpty(stageServiceUri))
+            {
+                throw new ArgumentNullException(nameof(stageServiceUri));
+            }
+
+            if (httpSource == null)
+            {
+                throw new ArgumentNullException(nameof(httpSource));
+            }
+
+            _stageServiceUri = stageServiceUri;
+            _httpSource = httpSource;
+        }
+
+        public async Task<StageListView> Create(string displayName,string apiKey, Common.ILogger log)
+        {
+            var result = await _httpSource.ProcessResponseAsync(
+                () =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Post, new Uri(new Uri(_stageServiceUri), StagePath));
+                    request.Headers.Add(ApiKeyHeader, apiKey);
+
+                    request.Content = new StringContent("\"" + displayName + "\"", Encoding.UTF8);
+                    request.Content.Headers.ContentType.MediaType = MediaType;
+
+                    return request;
+                },
+                async response =>
+                {
+                    response.EnsureSuccessStatusCode();

```

Comment:
It looks like this exception will flow all the way to the top. You may want to wrap this in another exception with a friendly message about being unable to create the staging environment, and then the http message.